### PR TITLE
doc: Add the security overview

### PR DESCRIPTION
**Pull request by upils**

### CODE REVIEW

File: docs/explanation/security-overview.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.. _security-overview:
+
+Security overview
+==============================
+
+Overview of security aspects of imagecraft.
```
Comment (rkratky) [inline docs/explanation/security-overview.rst:6]:
```suggestion
Overview of security aspects of Imagecraft.
```

File: docs/explanation/security-overview.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Privileged execution
+--------------------
+ 
+``imagecraft`` may require **elevated permissions** to properly run (when using the destructive mode). It is recommended to use a dedicated building machine. Make sure ``imagecraft`` is installed from a trusted source, and the provided configuration is trusted. 
+
+
+Cryptography
+------------
+
+``imagecraft`` is a wrapper around several tools and mainly ``ubuntu-image``. The tool itself is not using cryptographic technologies. Please refer to the `ubuntu-image security overview`_ for details on cryptographic technologies usage in ``ubuntu-image``.
```
Comment (rkratky) [inline docs/explanation/security-overview.rst:18]:
```suggestion
``imagecraft`` is a wrapper around several tools and mainly ``ubuntu-image``. The tool itself does not use cryptographic technologies. Refer to the `ubuntu-image security overview`_ for details on cryptographic technologies used in ``ubuntu-image``.
```

File: docs/explanation/security-overview.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+Miscellaneous
+~~~~~~~~~~~~~
+
+* Secrets (passwords and hashes) can be present in the configuration files provided to ``imagecraft`` to build images. Specifically, in the ``ubuntu-bootstrap`` plugin configuration:
+
+  * In the ``extra-ppas`` customisation section, authentication tokens ``user:password`` can be defined to access private PPAs. These values are used to write the ``apt`` configuration without any treatment.
```
Comment (rkratky) [inline docs/explanation/security-overview.rst:26]:
```suggestion
  * In the ``extra-ppas`` customisation section, authentication tokens (``user:password``) can be defined to access private PPAs. These values are used to write the ``apt`` configuration without any treatment.
```

**Review by rkratky — APPROVED** [review]
A few nitpick. Pre-approved.

**Review by sil2100 — APPROVED** [review]
LGTM!

**Pull request merged**